packages-next/types/src/index.ts

```diff
@@ -161,7 +161,7 @@ export type SessionContext = {
 
 export type KeystoneContext = {
   schemaName: 'public';
-  lists: any; // TODO: type this (itemAPI),
+  lists: Record<string, KeystoneListsAPI<Record<string, BaseGeneratedListTypes>>[string]>;
```

Comment:
This is a bit weird, looks like you're unwrapping and then rewrapping a record here?

Reply:
Oh, quite possibly, this was a straight cut and paste from somewhere I don't remember, TBH I didn't actually engage my brain on what the actual type is 😅 What would be a cleaner type here?

Reply:
Without reading the surrounding code and without being 100% sure, I _think_ that this is equivalent:

```
lists: KeystoneListsAPI<Record<string, BaseGeneratedListTypes>>;
```

You're basically accessing a property referenced by `string` in a Record, then wrapping that in a Record indexed by `string`, the line above removes both.

But also I don't think this is happening the right way, because the argument to `KeystoneListsAPI` should be defaulted as you have it, and is only needed if you're using the more specific generated types. Which is where this gets tricky.

Reply:
After much mucking around I'm 99% sure that the type as given is correct. It mirrors what we already have in `itemAPI.js`, and while I'm still getting the hang of these advanced types I don't think there's any alternative that makes everything work. The suggested equivalent did not work.

Reply:
```suggestion
  lists: KeystoneListsAPI<any>;
```